std/compress/io.go

```diff
@@ -0,0 +1,138 @@
+package compress
+
+import (
+	"errors"
+	"github.com/consensys/gnark/frontend"
+	"github.com/consensys/gnark/std/compress/internal/plonk_helpers"
+	"github.com/consensys/gnark/std/hash/mimc"
+	"github.com/consensys/gnark/std/lookup/logderivlookup"
+	"hash"
+	"math/big"
+)
+
+// Pack packs the words as tightly as possible, and works Big Endian: i.e. the first word is the most significant in the packed elem
```

Comment:
add in doc that this does not range check the input. Generally speaking for the io function that are exposed by the package, I think it's a bit confusing as to what is and what is not ranged checked.